core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -66,34 +70,77 @@ class WorkflowCompiler(
   def compile(
       logicalPlanPojo: LogicalPlanPojo,
       opResultStorage: OpResultStorage,
-      lastCompletedExecutionLogicalPlan: Option[LogicalPlan] = Option.empty,
       executionStateStore: ExecutionStateStore
   ): Workflow = {
+    // generate a LogicalPlan. The logical plan is the injected with all necessary sinks
+    val logicalPlan = compileLogicalPlan(logicalPlanPojo, executionStateStore)
 
-    // generate an original LogicalPlan. The logical plan is the injected with all necessary sinks
-    //  this plan will be compared in subsequent runs to check which operator can be replaced
-    //  by cache.
-    val originalLogicalPlan = compileLogicalPlan(logicalPlanPojo, executionStateStore)
-
-    // the cache-rewritten LogicalPlan. It is considered to be equivalent with the original plan.
-    val rewrittenLogicalPlan = WorkflowCacheRewriter.transform(
+    // assign the storage location to sink operators
+    assignSinkStorage(
+      logicalPlan,
       context,
-      originalLogicalPlan,
-      lastCompletedExecutionLogicalPlan,
-      opResultStorage,
-      logicalPlanPojo.opsToReuseResult.map(idString => OperatorIdentity(idString)).toSet
+      opResultStorage
     )
 
     // the PhysicalPlan with topology expanded.
-    val physicalPlan = PhysicalPlan(context, rewrittenLogicalPlan)
+    val physicalPlan = PhysicalPlan(context, logicalPlan)
 
     Workflow(
       context,
-      originalLogicalPlan,
-      rewrittenLogicalPlan,
```

Comment:
I merely remember there are some components afterwards using the returned `rewrittenLogicalPlan` just want to make sure we cleaned all usages.

Reply:
The only usage is `ExecutionReconfigurationService.modifyOperatorLogic`, by `val currentOp = workflow.logicalPlan.getOperator(opId)`. We remove the `rewrittenLogicalPlan` from the `Workflow`, so this caller will get the operator from the logicalPlan directly, which I believe is fine.